src/sentry/tasks/post_process.py

```diff
@@ -990,6 +990,12 @@ def _get_replay_id(event):
 
 
 def process_rules(job: PostProcessJob) -> None:
+    from sentry.buffer.redis import BufferHookEvent, BufferHookRegistry
+    from sentry.rules.processing.delayed_processing import process_delayed_alert_conditions
+
+    if not BufferHookRegistry._registry.get(BufferHookEvent.FLUSH):
+        BufferHookRegistry.add_callback(BufferHookEvent.FLUSH, process_delayed_alert_conditions)
```

Comment:
It's probably better to not expose the internal `_registry` here if possible, instead we could create a quick wrapper to validate this for us:
```suggestion
    if not redis_buffer_registry.has(BufferHookEvent.FLUSH):
        redis_buffer_registry.add_callback(BufferHookEvent.FLUSH, process_delayed_alert_conditions)
```

Then can move what you have into the function.
```python
def has(self, evt):
    return self._registry.get(evt) != None
```

--- 

This is also nice to have incase we want to refactor the registry later to allow for multiple event handlers.